File: lint-arwaky/archive/taxonomy/layer_names_vo.py
"""layer_names — Standardized architectural layer names."""

from .layer_content_vo import LayerNameVO

# Core Layers
LAYER_AGENT = LayerNameVO(value="agent")
LAYER_CAPABILITIES = LayerNameVO(value="capabilities")
LAYER_TAXONOMY = LayerNameVO(value="taxonomy")
LAYER_CONTRACT = LayerNameVO(value="contract")
LAYER_INFRASTRUCTURE = LayerNameVO(value="infrastructure")
LAYER_SURFACES = LayerNameVO(value="surfaces")
LAYER_ROOT = LayerNameVO(value="root")

# Global Scope
LAYER_GLOBAL = LayerNameVO(value="global")

ALL_CORE_LAYERS = [
    LAYER_AGENT,
    LAYER_CAPABILITIES,
    LAYER_TAXONOMY,
    LAYER_CONTRACT,
    LAYER_INFRASTRUCTURE,
    LAYER_SURFACES,
    LAYER_ROOT,
]

CORE_LAYER_NAMES = {str(layer) for layer in ALL_CORE_LAYERS}

File: lint-arwaky/src-rust/taxonomy/layer_names_vo.rs
use super::*;

pub fn layer_agent() -> LayerNameVO { LayerNameVO::new("agent") }
pub fn layer_capabilities() -> LayerNameVO { LayerNameVO::new("capabilities") }
pub fn layer_taxonomy() -> LayerNameVO { LayerNameVO::new("taxonomy") }
pub fn layer_contract() -> LayerNameVO { LayerNameVO::new("contract") }
pub fn layer_infrastructure() -> LayerNameVO { LayerNameVO::new("infrastructure") }
pub fn layer_surfaces() -> LayerNameVO { LayerNameVO::new("surfaces") }
pub fn layer_root() -> LayerNameVO { LayerNameVO::new("root") }
pub fn layer_global() -> LayerNameVO { LayerNameVO::new("global") }

pub fn all_core_layers() -> Vec<LayerNameVO> {
    vec![
        layer_agent(),
        layer_capabilities(),
        layer_taxonomy(),
        layer_contract(),
        layer_infrastructure(),
        layer_surfaces(),
        layer_root(),
    ]
}

pub fn core_layer_names() -> std::collections::HashSet<String> {
    all_core_layers().iter().map(|l| l.value.clone()).collect()
}
